lib/src/voip/models/participant.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:matrix/matrix.dart';
+
+class Participant {
+  final String userId;
+  final String deviceId;
+
+  Participant({required this.userId, required this.deviceId});
+
+  String get id => '$userId:$deviceId';
+
+  @override
+  String toString() {
+    return id;
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is Participant &&
+          userId == other.userId &&
+          deviceId == other.deviceId;
+
+  @override
+  int get hashCode => userId.hashCode ^ deviceId.hashCode;
+
+  factory Participant.fromId(String id) {
+    final int lastIndex = id.lastIndexOf(':');
```

Comment:
Are you sure you can do that? Nothing prevents you from having a : in the deviceId afaik.